agenta-backend/agenta_backend/services/evaluators_service.py

```diff
@@ -909,24 +1141,33 @@ def auto_similarity_match(
         )
 
 
+async def similarity_match(input: EvaluatorInputInterface) -> EvaluatorOutputInterface:
+    set1 = set(input.inputs["prediction"].split())
+    set2 = set(input.inputs["ground_truth"].split())
+    intersect = set1.intersection(set2)
+    union = set1.union(set2)
+
+    similarity = len(intersect) / len(union)
+    is_similar = True if similarity > input.settings["similarity_threshold"] else False
+    return {"outputs": {"success": is_similar}}
+
+
 async def semantic_similarity(
-    output: Union[str, Dict[str, Any]],
-    correct_answer: str,
-    api_key: str,
-) -> float:
-    """
-    Calculate the semantic similarity score of the LLM app using OpenAI's Embeddings API.
+    input: EvaluatorInputInterface,
+) -> EvaluatorOutputInterface:
+    """Calculate the semantic similarity score of the LLM app using OpenAI's Embeddings API.
 
     Args:
-        output (str): the output text
-        correct_answer (str): the correct answer text
+        input (EvaluatorInputInterface): the evaluator input
 
     Returns:
         float: the semantic similarity score
     """
-    if not isinstance(output, str):
-        output = output.get("data", "")
 
+    if not isinstance(input.inputs["prediction"], str):
+        output = input.inputs["prediction"].get("data", "")
```

Comment:
These checks should be kept outside of the core evaluator, within the legacy part. That applies to all evaluators.

Reply:
True, thank you.